app/src/main/java/com/sunasterisk/thooi/data/source/remote/PostRemoteDataSource.kt

```diff
@@ -0,0 +1,80 @@
+package com.sunasterisk.thooi.data.source.remote
+
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+import com.sunasterisk.thooi.data.Result
+import com.sunasterisk.thooi.data.source.entity.Post
+import com.sunasterisk.thooi.data.source.PostDataSource
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.tasks.await
+
+class PostRemoteDataSource(database: FirebaseFirestore) : PostDataSource.Remote {
```

Comment:
```suggestion
class PostRemoteDataSource(
    database: FirebaseFirestore,
    private val dispatcher = Dispatchers.IO
) : PostDataSource.Remote {
```
`inject` vào `constructor` để viết test